examples/amm/src/contract.rs

```diff
@@ -0,0 +1,342 @@
+                    .await?;
+                self.send_to(&owner, output_token_idx, output_amount).await
+            }
+            OperationType::AddLiquidity {
+                owner,
+                token0_amount,
+                token1_amount,
+            } => {
+                if token0_amount == Amount::ZERO || token1_amount == Amount::ZERO {
+                    return Err(AmmError::NoZeroAmounts);
+                }
+
+                let balance0 = self.get_pool_balance(0).await?;
+                let balance1 = self.get_pool_balance(1).await?;
+
+                if balance0 > Amount::ZERO && balance1 > Amount::ZERO {
+                    let one_amount_bigint =
+                        BigUint::from_u128(<Amount as Into<u128>>::into(Amount::ONE))
+                            .expect("Couldn't generate Amount::ONE in bigint");
+
+                    let balance0_bigint =
+                        BigUint::from_u128(<Amount as Into<u128>>::into(balance0))
+                            .expect("Couldn't generate balance0 in bigint");
+                    let balance1_bigint =
+                        BigUint::from_u128(<Amount as Into<u128>>::into(balance1))
+                            .expect("Couldn't generate balance1 in bigint");
+                    let token0_amount_bigint =
+                        BigUint::from_u128(<Amount as Into<u128>>::into(token0_amount))
+                            .expect("Couldn't generate token0_amount in bigint");
+                    let token1_amount_bigint =
+                        BigUint::from_u128(<Amount as Into<u128>>::into(token1_amount))
+                            .expect("Couldn't generate token1_amount in bigint");
+
+                    // To always maintain the precision, whenever we divide,
+                    // we always multiply by Amount::ONE before dividing
+                    let old_ratio_bigint =
+                        (balance0_bigint * one_amount_bigint.clone()) / balance1_bigint;
+                    let new_ratio_bigint =
+                        (token0_amount_bigint * one_amount_bigint) / token1_amount_bigint;
+
+                    if balance1 > Amount::ZERO && old_ratio_bigint != new_ratio_bigint {
+                        return Err(AmmError::BalanceRatioAlteredError);
+                    }
+                }
+
+                self.receive_from_account(&owner, 0, token0_amount).await?;
+                self.receive_from_account(&owner, 1, token1_amount).await?;
+
+                Ok(())
+            }
+            // When removing liquidity, you'll specify one of the tokens you want to
+            // remove and the amount, and we'll calculate the amount for the other token that
+            // we'll remove based on the current ratio, and remove them.
+            OperationType::RemoveLiquidity {
+                owner,
+                input_token_idx,
+                input_amount,
```

Comment:
```suggestion
                output_token_idx,
                output_amount,
```
Or just `token_idx` and `amount`?